posts/2021-09-03-microprofile-21009.adoc

```diff
@@ -0,0 +1,268 @@
+...
+    @Override
+     public HealthCheckResponse call() {
+       ...
+       if (getMemUsage() < 0.95) {
+           return HealthCheckResponse.up("memUsage");
+       }
+       return HealthCheckResponse.down("memUsage");
+       ...
+     }
+}
+----
+
+To view the status of each health check, access the `http://<hostname>:<port>/health/started` endpoint.
+
+
+To enable the new beta features in your app, add them to your `server.xml`:
+
+[source, xml]
+----
+<feature>mpHealth-3.1</feature>
+----
+
+To find out more visit:
+
+* link:https://github.com/eclipse/microprofile-health/releases/tag/3.1[MicroProfile Health Check 3.1 Release Page]
+* link:https://github.com/eclipse/microprofile-health/blob/3.1/spec/src/main/asciidoc/release_notes.asciidoc[Release Notes]
+
+To get started with and learn more about MicroProfile Health, check out our link:https://openliberty.io/guides/microprofile-health.html[MicroProfile Health guide].
+
+
+[#transaction]
+== Transaction Peer Recovery
+
+This update provides autonomous transactional peer recovery among groups of peer application servers either in a cloud or otherwise. This will allow 2 phase commit to be used in cloud environments for the first time and will enable the clearing of database locks held by servers that have been "reaped" by cloud runtimes. 
+
+In a cloud environment such as OpenShift, Open Liberty servers can be dynamically created or deleted, for example, to handle variations in system load. This possibility poses a problem for applications that use transactions. The sudden removal of a server instance might occur during two-phase commit (2PC) processing and leave transactional resources locked. This update alleviates this problem by allowing the configuration of Open Liberty servers to automatically recover transactions on behalf of other servers. This process is called peer recovery. 
+
+You can configure it by specifying the `recoveryGroup` and `recoveryIdentity` attributes in the `transaction` stanza of your `server.xml` file, as shown in the following example:
+
+[source, xml]
+----
+<transaction
+...
+recoveryGroup="peer-group-name"
+recoveryIdentity="${HOSTNAME}${wlp.server.name}"
+...
+/>
+----
+
+For more information regarding the Transaction Manager service, reference the server configuration link:{url-prefix}/docs/latest/reference/config/transaction.html[docs for transaction].
+
+
+[#tbasic]
+== TBASIC Logging Format
+
+A new logging format, `TBASIC`, is added to match the Websphere Application Server basic logging format. This new format option can be used for `consoleFormat`, `messageFormat`, or `traceFormat`. `TBASIC` format matches the existing `BASIC` format for `traceFormat` (so now either `TBASIC` or `BASIC` can be used to refer to this format for `traceFormat`). Using this new format option will allow users that utilize log parsers that work with the Websphere Application Server basic format to use the same parsers with Open Liberty logs.
```

Comment:
Change this paragraph to:

A new logging format `TBASIC` has been made available for both `consoleFormat` and `messageFormat`.  Although a `BASIC` logging format already existed for `consoleFormat`, `messageFormat`, and `traceFormat` the output differed between `traceFormat` and `consoleFormat` and `messageFormat`.  The new `TBASIC` format has been added to match the existing `BASIC` format that was already available for `traceFormat`, providing the ability to have consistent formatting across all three logs.  The `TBASIC` format has also been added as a logging format for `traceFormat`, acting as an alias for the existing `BASIC` option.
To summarize, if you want to have the same logging format in `consoleFormat` and `messageFormat` that you could get in `traceFormat` using `BASIC`, you now can by using the `TBASIC` logging format.